<launch>

	<!-- arduino communication node -->
	<node respawn="true" pkg="rosserial_arduino" type="serial_node.py" name="serial_node" >
		<param name="port" type="string" value="/dev/ttyUSB0" />
		<param name="deadzone" value="0.05" />
	</node>
	
	<!-- joy node -->
	<node respawn="true" pkg="joy" type="joy_node" name="joy_node" >
		<param name="dev" type="string" value="/dev/input/js0" />
		<param name="deadzone" value="0.05" />
	</node>

	<node pkg="lunabotics2020" type="DriveController" name="DriveController" />

	<node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
		<param name="serial_port"         type="string" value="/dev/ttyUSB1"/>  
		<param name="serial_baudrate"     type="int"    value="256000"/>
		<param name="frame_id"            type="string" value="beacon_frame"/>
		<param name="inverted"            type="bool"   value="false"/>
		<param name="angle_compensate"    type="bool"   value="true"/>
		<param name="scan_mode"			  type="string" value="DenseBoost"/>

		<!--Change topic name to beacon_scan-->
		<remap from="scan" to="beacon_scan"/>
	</node>

	<node pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen" name="laser_filter">
		<remap from="scan" to="beacon_scan" />  <!--Topic to subscribe to-->
		<remap from="scan_filtered" to="beacon_scan_filtered"/>  <!--Topic to publish to-->

		<!--Load config file-->
		<rosparam command="load" file="$(find lunabotics2020)/launch/beacon_filter.yaml" />
	</node>

	<node pkg="lunabotics2020" type="FlagLocalization" name="FlagLocalization"></node>

	<!--node pkg="lunabotics2020" type="Odometry" name="Odometry"></node-->

  <!--Transforms-->

	<node pkg="tf2_ros" type="static_transform_publisher" name="beacon_broadcaster" args="3.255 0.04 0 0 0 0 map beacon_frame" /> <!-- Map to Lidar Transform-->
	<node pkg="tf2_ros" type="static_transform_publisher" name="flag_broadcaster" args="-.15 .3 0 0 0 0 flag transportation" /> <!-- Flag to Center of Robot Transform-->

  	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find lunabotics2020)/.rviz/default.rviz" />

</launch>
